app/src/main/java/org/systers/mentorship/view/activities/SignUpActivity.kt

```diff
@@ -0,0 +1,132 @@
+package org.systers.mentorship.view.activities
+
+import android.arch.lifecycle.Observer
+import android.arch.lifecycle.ViewModelProviders
+import android.os.Bundle
+import android.support.design.widget.Snackbar
+import android.widget.Toast
+import kotlinx.android.synthetic.main.activity_sign_up.*
+import org.systers.mentorship.R
+import org.systers.mentorship.remote.responses.RegisterRequest
+import org.systers.mentorship.viewmodels.SignUpViewModel
+
+/**
+ * This activity will let the user to sign up into the system using name, username,
+ * email and password.
+ */
+class SignUpActivity : BaseActivity() {
+
+    private lateinit var signUpViewModel: SignUpViewModel
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_sign_up)
+        signUpViewModel  = ViewModelProviders.of(this).get(SignUpViewModel::class.java)
+        signUpViewModel.successful.observe(this, Observer {
+            successful ->
+            hideProgressDialog()
+            if (successful != null) {
+                if (successful) {
+                    Toast.makeText(this, signUpViewModel.message, Toast.LENGTH_LONG)
+                            .show()
+                    onBackPressed()
```

Comment:
Don't use `onBackPressed()`. Finish the previous activity and start the next activity using an intent.

Also, include a button in this layout to go to LoginActivity.

Reply:
its included, I'll show the UI after the review fixes are complete